sources/packages/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -673,17 +707,208 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
+    );
+  }
+
+  /**
+   * Approve or Decline an offering change
+   * requested by institution.
+   * @param offeringId offering that is requested for change.
+   * @param userId User who approves or declines the offering.
+   * @param assessmentNotes Notes added during the process.
+   * @param offeringStatus Approval or Decline status.
+   */
+  async assessOfferingChangeRequest(
+    offeringId: number,
+    userId: number,
+    assessmentNotes: string,
+    offeringStatus: OfferingStatus,
+  ): Promise<void> {
+    const requestedOffering = await this.getOfferingSummaryToRequestChange(
+      offeringId,
+      OfferingStatus.AwaitingApproval,
+    );
+
+    if (!requestedOffering) {
+      throw new CustomNamedError(
+        "Either offering not found or the offering not in appropriate status to be approved or declined for change.",
+        OFFERING_NOT_VALID,
+      );
+    }
+
+    if (!requestedOffering.precedingOffering) {
+      throw new CustomNamedError(
+        "The offering requested for change does not have a preceding offering.",
+        OFFERING_NOT_VALID,
+      );
+    }
+    const precedingOffering = requestedOffering.precedingOffering;
+    const auditUser = { id: userId } as User;
+    const currentDate = new Date();
+    // Set audit fields.
+    requestedOffering.modifier = auditUser;
+    requestedOffering.updatedAt = currentDate;
+    requestedOffering.assessedBy = auditUser;
+    requestedOffering.assessedDate = currentDate;
+
+    precedingOffering.modifier = auditUser;
+    precedingOffering.updatedAt = currentDate;
+    precedingOffering.assessedBy = auditUser;
+    precedingOffering.assessedDate = currentDate;
+    let applications: ApplicationAssessmentSummary[] = [];
+
+    // Populate the status accordingly and get impacted applications
+    // when the offering change is approved.
+    if (offeringStatus === OfferingStatus.Approved) {
+      requestedOffering.offeringStatus = OfferingStatus.Approved;
+      precedingOffering.offeringStatus = OfferingStatus.ChangeOverwritten;
+      applications = await this.getApplicationsToSubmitReassessment(offeringId);
+
+      for (const application of applications) {
+        application.currentAssessment = {
+          application: { id: application.id } as Application,
+          triggerType: AssessmentTriggerType.OfferingChange,
+          offering: { id: offeringId } as EducationProgramOffering,
+          creator: auditUser,
+          createdAt: currentDate,
+          submittedBy: auditUser,
+          submittedDate: currentDate,
+        } as StudentAssessment;
+        application.modifier = auditUser;
+        application.updatedAt = currentDate;
+      }
+    } else {
+      requestedOffering.offeringStatus = OfferingStatus.ChangeDeclined;
+      precedingOffering.offeringStatus = OfferingStatus.Approved;
+    }
+
+    await this.dataSource.transaction(async (transactionalEntityManager) => {
+      // create the note for assessment.
```

Comment:
Please ensure upper case on these comments and the below ones.
@ann-aot @guru-aot @henryoforeh-aot we have a helper method to create notes for students. Next time that someone needs to create an institution note we can have the same for the institution also.
It will simplify the line 835 till 854 to something like below.
![image](https://user-images.githubusercontent.com/61259237/182470247-115eb876-2c5b-4522-af99-8dae9cc7f911.png)

Reply:
Yes it would be better to create something for Institution note too.